R/impute.R

```diff
@@ -4,53 +4,59 @@
-#' then constructed based upon the method as specified in the call to [draws()] with 1 dataset being generated per
-#' set of parameter estimates provided by [draws()].
-#'
-#' - Bayes & Approximate Bayes: each imputed dataset consists of 1 row per subject & visit from the original
-#' dataset with missing values imputed by taking a single random sample from the augmented outcome distribution.
-#'
-#' - Conditional Mean: each imputed dataset consists of 1 row per subject & visit from the bootstrapped or Jackknife
-#' dataset that was used to generate the corresponding parameter estimates in [draws()].
-#' Missing values are imputed by using the mean of the augmented outcome distribution.
-#'
-#' For conditional mean and approximate Bayes the first imputed dataset will be the original dataset imputed
-#' using the ML parameter estimates of the
-#' MMRM model fitted to the original dataset. See TODO (reference paper for methods) for full details.
-#'
-#'
-#' `update_strategy` can be used to update the imputation method that was originally set via
-#' the `data_ice` option in `draws`.
-#' The `update_strategy` dataframe must contain two columns, one for the subject ID and another for the imputation method,
-#' whose names are the same as
-#' those defined in `vars` argument as specified in the call to `draws`.
-#' You can only update the imputation method, that is to say that specifying
-#' a visit column has no effect here. If you wish to change the ICE visit you will need to re-run `draws`.
-#' A key limitation is that you can only switch between a MAR and a non-MAR method (or vice versa)
-#' if that subject has no non-missing post ICE data.
-#' As an example, if a subject had their ICE on "Visit 2" but still had values in the dataset for
-#' "Visit 3" then the function will throw and error if you try to switch them from MAR to a non-MAR method or
-#' from a non-MAR method to MAR.
+#' `impute()` works by taking the model parameter estimates, as generated by [draws()], and using
+#' them to calculate the conditional distribution for the outcome for each subject separately.
+#' These distributions are then augmented based upon the; intercurrent event (ICE) data,
+#' the chosen imputation strategy, and the chosen reference group. Imputed values are then drawn
+#' from the augmented imputation distribution with 1 dataset being generated per set of parameter
+#' estimates provided by [draws()].
+#'
+#' The exact manor in how missing values are imputed from this augmented distribution depends
+#' on the method object that was provided to [draws()], in particular:
+#'
+#' - Bayes & Approximate Bayes: each imputed dataset consists of 1 row per subject & visit
+#' from the original dataset with missing values imputed by taking a single random sample
+#' from the augmented outcome distribution.
+#'
+#' - Conditional Mean: each imputed dataset consists of 1 row per subject & visit from the
```

Comment:
Maybe we should mention here that also the original dataset is imputed at this stage?

Reply:
Not quite sure what you mean ?

Reply:
I mean, we don't just impute each bootstrap/jackknife sample but also the full original dataset. Maybe it is worthed to mention this here.

Reply:
Yer I wasn't sure about this. I had originally included a line but then removed it as I was wondering if we were duplicating text from both `draws()` and `methods()`.  If you think it still adds value I am happy to add it back in

Reply:
done